tests/model_registry/conftest.py

```diff
@@ -303,3 +305,80 @@ def generated_schema(model_registry_instance_rest_endpoint: str) -> Any:
         uri="https://raw.githubusercontent.com/kubeflow/model-registry/main/api/openapi/model-registry.yaml",
         base_url=f"https://{model_registry_instance_rest_endpoint}/",
     )
+
+
+@pytest.fixture(scope="class")
+def dsc_resource_scope_class(admin_client: DynamicClient) -> DataScienceCluster:
+    dsc = DataScienceCluster(name=py_config["dsc_name"], ensure_exists=True)
+    return dsc
+
+
+@pytest.fixture(scope="class")
+def updated_dsc_component_state_scope_class(
+    request: FixtureRequest,
+    model_registry_namespace: Namespace,
+    dsc_resource_scope_class: DataScienceCluster,
+) -> Generator[DataScienceCluster, Any, Any]:
+    original_components = dsc_resource_scope_class.instance.spec.components
+    with ResourceEditor(patches={dsc_resource_scope_class: {"spec": {"components": request.param["component_patch"]}}}):
```

Comment:
`"managementState": DscComponents.ManagementState.MANAGED` - this is set in all tests - set it once here

the component is always `DscComponents.MODELREGISTRY` so set it here, no need to pass

Reply:
I would not hardcode anything. This gives me flexibility to pass different combinations for both `registriesNamespace` and `managementState`. We don't yet know if any other settings can be supported in future.